odoo/fields.py

```diff
@@ -25,6 +26,11 @@
 from psycopg2.extras import Json as PsycopgJson
 from difflib import get_close_matches, unified_diff
 from hashlib import sha256
+from typing import Any, TypeVar, Generic, ForwardRef, get_args, overload
+try:
```

Comment:
For some typing information, you could import them `if typing.TYPE_CHECKING` reducing the footprint of what is loaded at runtime. You should do this especially for `Self`.

(Same in other files)